Documentation/Yarn-Spec.md

```diff
@@ -0,0 +1,636 @@
+
+The _keyword_ is the text `set`.
+The _variable_ is the name of the variable which is to have its value changed.
+The _operator_ must be the text `to`.
+The _expression_ is any expression.
+
+The following is an example of two set commands:
+```yarn
+<<set $name to "General Kenobi">>
+<<set $boldness to $boldness + 1>>
+```
+
+The set command must follow all the rules for variable naming and expressions.
+The set command must not allow setting a variable to an expression who's value is different from the type of that variable.
+
+### Flow control
+
+Flow control is a collection of commands that allow the writer to control the flow of the story.
+There are four commands which work in conjunction to support flow control:
+
+- if
+- else
+- elseif
+- endif
+
+The order of these commands is always the same and must be followed:
+
+1. if
+1. elseif
+1. else
+1. endif
+
+The if and endif must be present, the elseif and else must be optional.
+While each of these commands are their own statement they should be considered to be part of a larger flow control statement which spans multiple lines.
+
+#### if
+
+The _if_ command is the opening command of flow control and is broken up into two parts, the keyword and the expression and must be in that order.
+The _keyword_ is the text `if`.
+The _expression_ is an expression.
+The expression must resolve to a boolean.
+
+`<<if $boldness > 1>>` is an example of an if command, `<<if 1>>` is an example of an invalid if, it is invalid because the expression does not resolve to a boolean.
+
+#### elseif
+
+The _elseif_ command is an optional component of flow control and allows for additional flow to be expressed.
+The command works in a fashion very similar to the if command.
+The command is broken up into two parts, the keyword and the expression and must be presented in that order.
+The _keyword_ is the text `elseif`.
+The _expression_ is an expression.
+The expression must resolve to a boolean.
+
+The minimum mumber of required elseif commands must be zero.
+The maximum number of allowed elseif commands is unspecified but should be greater than zero.
+An elseif command must not exist without an if command and must go after the if command.
+
+#### else
+
+The _else_ command is an optional component of flow control and allows for additional flow to be expressed.
+The command only has a single component, the keyword `else`.
+
+There must only be a single else command (if any) per flow control.
+The else command must go after the if and any elseif commands.
+The else must not exist without an if command.
+
+The example of the else command is `<<else>>`.
+
+#### endif
+
+The _endif_ command is the final element of flow control and is comprised solely of the keyword `endif`.
+The endif must be present whenever there is flow control and must go after the if and any elseif or else commands.
```

Comment:
"must go after the if and any elseif or else commands" makes it sound like `endif` is required at the end of each and every branch. Recommend "must go after the final if/elseif/else command of a flow control block"